CMakeLists.txt

```diff
@@ -1,58 +1,201 @@
-option (ENABLE_HTML_DOCS "Build MongoDB C Driver HTML documentation." OFF)
-option (ENABLE_EXTRA_ALIGNMENT
-   "Turn on extra alignment of libbson types. Set to ON/OFF, default ON.\
- Required for the 1.0 ABI but better disabled."
-   ON
+cmake_minimum_required (VERSION 3.15)
+
+project (mongo-c-driver C)
+
+list (APPEND CMAKE_MODULE_PATH
+      "${PROJECT_SOURCE_DIR}/build/cmake"
+      "${PROJECT_SOURCE_DIR}/build/cmake/make_dist"
+      )
+
+include (MongoSettings)
+
+# Subcomponents:
+mongo_bool_setting(ENABLE_MONGOC "Enable the build of libmongoc libraries (The MongoDB C database driver)")
+mongo_bool_setting(
+   USE_SYSTEM_LIBBSON [[Use an existing libbson library (via find_package()) instead of building one from source]]
+   DEFAULT VALUE OFF
+   ADVANCED
 )
-option (ENABLE_RDTSCP
-   "Fast performance counters on Intel using the RDTSCP instruction"
-   OFF
+
+# Deprecated option "ENABLE_BSON" is gone, but old caches may rely on SYSTEM to
+# disable the libbson build. Allow ENABLE_BSON=SYSTEM only if USE_SYSTEM_LIBBSON
+# is also true, to allow both ENABLE_BSON=SYSTEM and USE_SYSTEM_LIBBSON=TRUE to
+# simplify downstream build scripts
+if(ENABLE_BSON STREQUAL "SYSTEM" AND NOT USE_SYSTEM_LIBBSON)
+   # User set ENABLE_BSON=SYSTEM, so they intend for us to build against an external
+   # libbson, but they didn't set USE_SYSTEM_LIBBSON=ON, so they are probably
+   # unaware of this change. Rather than miscompile, stop what we're doing:
+   message(FATAL_ERROR "ENABLE_BSON=SYSTEM is no longer supported. Use USE_SYSTEM_LIBBSON=ON for the same effect.")
+endif()
+
+# Static libraries:
+mongo_setting(
+   ENABLE_STATIC "Enable building and installing static library archive artifacts"
+   OPTIONS ON OFF BUILD_ONLY
+   DEFAULT VALUE "ON"
 )
-option (ENABLE_APPLE_FRAMEWORK "Build libraries as frameworks on darwin platforms" OFF)
-set (ENABLE_ICU AUTO CACHE STRING "Enable ICU support, necessary to use non-ASCII usernames or passwords, default AUTO.")
-option (ENABLE_UNINSTALL "Enable creation of uninstall script and associated uninstall build target." ON)
-set (ENABLE_CLIENT_SIDE_ENCRYPTION AUTO CACHE STRING "Enable In-Use Encryption support. Requires libmongocrypt. Set to ON/AUTO/OFF, default AUTO.")
-set (ENABLE_MONGODB_AWS_AUTH AUTO CACHE STRING "Enable support for MONGODB-AWS authentication mechanism. Set to ON/AUTO/OFF, default AUTO. (also requires ENABLE_SSL not set to OFF)")
-option (ENABLE_PIC
-   "Enables building of position independent code for static library components."
-   ON
+if(NOT WIN32)
+   mongo_bool_setting(ENABLE_PIC "Enable position-independent code when building static libraries (Not applicable on Windows)")
+elseif(DEFINED ENABLE_PIC)
+   # We are on Windows, and ENABLE_PIC is defined.
+   message(STATUS "ENABLE_PIC=“${ENABLE_PIC}” has no effect on Windows (All Windows code is position-independent)")
+else()
+   # We are on Windows, and ENABLE_PIC is not defined, which is normal
+endif()
```

Comment:
Suggest using pattern consistent with new MongoSettings utilities, e.g.
```cmake
mongo_bool_setting(
   ENABLE_PIC "Enable position-independent code when building static libraries"
   DEFAULT EVAL [[
      if (WIN32)
         set(DEFAULT OFF)
      else ()
         set(DEFAULT ON)
      endif ()
   ]]
   VALIDATE CODE [[
      if (WIN32 AND DEFINED ENABLE_PIC)
         message(NOTICE "ENABLE_PIC="${ENABLE_PIC}" has no effect on Windows")
      endif ()
   ]]
)
```

Reply:
The conditional is to not even define the setting on Windows, and detect if it has been explicitly set by the user, which means their explicit setting is being ignored. The suggested replacement would unconditionally issue the NOTICE on Windows since the option will always be defined when going through that code path.

Reply:
It would be re-defined every time configure passes through `mongo_setting()`